pkg/client/messaging/client.go

```diff
@@ -95,6 +105,24 @@ func SendByDestination(destination *service.Destination) SendMessageOpions {
 	}
 }
 
+// WaitForResponse option to set message response type.
+// Message reply will wait for the response of this message type and matching thread ID.
+func WaitForResponse(responseType string) SendMessageOpions {
+	return func(opts *sendMsgOpts) {
+		opts.responseMsgType = responseType
+	}
+}
+
+// ResponseTimeout is timeout waiting for response.
+func ResponseTimeout(timeout time.Duration) SendMessageOpions {
```

Comment:
should use go context style, not durations.

Reply:
done